menas/src/main/scala/za/co/absa/enceladus/menas/services/DatasetService.scala

```diff
@@ -102,4 +101,150 @@ class DatasetService @Autowired() (datasetMongoRepository: DatasetMongoRepositor
     }
   }
 
+  override def importItem(item: Dataset, username: String): Future[Option[Dataset]] = {
+    getLatestVersionValue(item.name).flatMap {
+      case Some(version) => update(username, item.copy(version = version))
+      case None => super.create(item.copy(version = 1), username)
+    }
+  }
+
+  override def validateSingleImport(item: Dataset, metadata: Map[String, String]): Future[Validation] = {
+    val confRulesWithConnectedEntities = item.conformance.filter(_.hasEntityConnected)
+    val maybeSchema = datasetMongoRepository.getConnectedSchema(item.schemaName, item.schemaVersion)
+
+    val validations = super.validateSingleImport(item, metadata)
+    val validationsWithSchema = validateSchema(item.schemaName, item.schemaVersion, validations, maybeSchema)
+    val validationWithConnected = validateConnectedEntities(confRulesWithConnectedEntities, validationsWithSchema)
+    validateConformanceRules(item.conformance, maybeSchema, validationWithConnected)
+  }
+
+  private def validateConnectedEntities(confRulesWithConnectedEntities: List[ConformanceRule],
+                                    validationWithSchema: Future[Validation]): Future[Validation] = {
+    confRulesWithConnectedEntities.foldLeft(validationWithSchema) { (acc, cr) =>
+      val maybeConnectedEntity = cr.maybeConnectedEntityType.get match {
+        case "mappingTable" => datasetMongoRepository.getConnectedMappingTable(
```

Comment:
This part of code seems fragile. At the very least, there should be a `case _ =>`, otherwise all future connected entity type would be looked up among mapping tables. If you make the entity type enum, you'll get exhaustion check here :) 

Another solution could probably have mappings for the appropriate connected types to their their validation routines.

Reply:
I tried something new. Please review

Reply:
Looks much better. When adding a new connected entity type, you will still need to add cases here, but at least you will be notified about this during compilation because sealed trait should be checked for exhaustion. So fine by me.